### DISCUSSION TITLE
set octave in note

### DISCUSSION THREAD

**Discussion by zina0316** [alda-lang/alda#466] [Ideas]
My idea:
```alda
 note name
 ↓
4c4 ← note length
↑
octave
```

**Comment by daveyarwood** [comment]
Hi, thanks for the suggestion! When I was designing the Alda language, I took inspiration from [Modern MML](https://en.wikipedia.org/wiki/Music_Macro_Language#Modern_MML), which keeps octaves separate from notes. I think it's cleaner that way, and the code ends up being more concise.

Compare:

```
4c4 4d4 4e4 4f4 4g4 4a4 4b4 5c4
```

with:

```alda
o4 c4 d4 e4 f4 g4 a4 b4 > c4
```

And for extra conciseness, both MML and Alda allow you to omit the duration from notes, and it will default to the last-used duration:

```alda
o4 c4 d e f g a b > c
```

**Reply by zina0316** [reply]
those two systems can work together
